src/components/Home.js

```diff
@@ -12,12 +15,50 @@ const Home = (props) => {
     history.push('/ListView');
   };
 
+  const onTokenInputChange = (e) => {
+    setToken(e.target.value);
+  };
+
+  const onSubmitHandler = (e) => {
+    e.preventDefault();
+
+    db.collection(token)
+      .get()
+      .then(function (querySnapshot) {
+        if (querySnapshot.empty) {
+          alert('uh oh! list does not exist');
+        } else {
+          localStorage.setItem('token', token);
+          history.push('/ListView'); // when joining url is /ListView but ListView component does not render until refresh
+        }
+      })
+      .catch(function (error) {
+        alert(error);
+      });
+
+    setToken('');
+  };
+
   return (
     <div className="home">
       <h1>Welcome to Smart Shopping List!</h1>
       <button className="new-list-btn" onClick={generateToken}>
         Create a new list
       </button>
+      <p>- or -</p>
+      <p>Join an existing shopping list by entering a three word token.</p>
+      <form onSubmit={onSubmitHandler}>
+        <label htmlFor="joinList">Share token </label>
```

Comment:
I might label this form "Join List" or "Paste Token Here" ? Since we are asking people to input a token, not generate one.

Reply:
We labeled it as such based on the wireframe provided, but I agree that "paste token here" would make more sense! updated here: https://github.com/the-collab-lab/tcl-16-smart-shopping-list/pull/23/commits/9f36b0863a7cf48f53d0f414fe4e3897d04e05af